include/xtensor/xtiny.hpp

```diff
@@ -0,0 +1,1999 @@
+    {
+        std::ignore = n;
+        XTENSOR_ASSERT_MSG(n == size(), "xtiny_impl::assign(n, v): size mismatch.");
+        base_type::fill(v);
+    }
+
+    template <class V, index_t N>
+    template <class IT, class>
+    inline void
+    xtiny_impl<V, N, std::array<V, (size_t)N>>::assign(IT begin, IT end)
+    {
+        std::ignore = end;
+        XTENSOR_ASSERT_MSG(std::distance(begin, end) == static_cast<std::ptrdiff_t>(size()),
+            "xtiny_impl::assign(begin, end): size mismatch.");
+        for(size_type k=0; k<N; ++k, ++begin)
+        {
+            (*this)[k] = static_cast<value_type>(*begin);
+        }
+    }
+
+    template <class V, index_t N>
+    constexpr inline auto
+    xtiny_impl<V, N, std::array<V, (size_t)N>>::capacity() const -> size_type
+    {
+        return N;
+    }
+
+    template <class V, index_t N>
+    constexpr inline auto
+    xtiny_impl<V, N, std::array<V, (size_t)N>>::shape() const -> index_type
+    {
+        return {N};
+    }
+
+    /**********************************************/
+    /* xtiny_impl fixed shape view implementation */
+    /**********************************************/
+
+    template <class V, index_t N, class R>
+    inline
+    xtiny_impl<V, N, R>::xtiny_impl()
+    : m_data()
+    {
+    }
+
+    template <class V, index_t N, class R>
+    inline
+    xtiny_impl<V, N, R>::xtiny_impl(representation_type const & begin)
+    : m_data(begin)
+    {
+    }
+
+    template <class V, index_t N, class R>
+    inline
+    xtiny_impl<V, N, R>::xtiny_impl(representation_type const & begin, representation_type const & end)
+    : m_data(begin)
+    {
+        std::ignore = end;
+        XTENSOR_ASSERT_MSG(std::distance(begin, end) == static_cast<std::ptrdiff_t>(size()),
+            "xtiny_impl(begin, end): size mismatch");
+    }
+
+    template <class V, index_t N, class R>
+    template <class IT, class>
+    inline
+    xtiny_impl<V, N, R>::xtiny_impl(IT begin, IT end)
+    : m_data(const_cast<representation_type>(&*begin))
+    {
+        std::ignore = end;
+        XTENSOR_ASSERT_MSG(std::distance(begin, end) == static_cast<std::ptrdiff_t>(size()),
+            "xtiny_impl::assign(begin, end): size mismatch.");
+    }
+
+    template <class V, index_t N, class R>
+    inline void
+    xtiny_impl<V, N, R>::reset(representation_type const & begin)
+    {
+        m_data = begin;
+    }
+
+    template <class V, index_t N, class R>
+    inline void
+    xtiny_impl<V, N, R>::assign(size_type n, const value_type& v)
+    {
+        std::ignore = n;
+        XTENSOR_ASSERT_MSG(n == size(), "xtiny_impl::assign(n, v): size mismatch.");
+        for(size_type k=0; k<N; ++k)
+        {
+            (*this)[k] = v;
+        }
+    }
+
+    template <class V, index_t N, class R>
+    template <class IT, class>
+    inline void
+    xtiny_impl<V, N, R>::assign(IT begin, IT end)
+    {
+        std::ignore = end;
```

Comment:
nice trick! this is to avoid unused parameter warnings, right?

Reply:
Yes.